lib/iris/experimental/ugrid.py

```diff
@@ -204,6 +211,58 @@ def validate_arg_vs_list(arg_name, arg, valid_list):
             attributes=attributes,
         )
 
+    def __repr__(self):
+        def kwargs_filter(k, v):
+            result = False
+            if k != "cf_role":
+                if v is not None:
+                    result = True
+                    if (
+                        not isinstance(v, str)
+                        and isinstance(v, Iterable)
+                        and not v
+                    ):
+                        result = False
+                    elif k == "units" and v == "unknown":
+                        result = False
```

Comment:
If the `units` are `unknown`, then don't print them... this is the `units` equivalent to being `None`